Feature/186 dropdown menu logout

GuillemBo opened the pull request:
Added dropdown menu with user functionality:

- Implemented a dropdown menu that displays the GitHub username and includes a logout button.

- Added language selection functionality to switch between languages.

- Developed a menu that works seamlessly on both desktop and mobile views.

GuillemBo committed:
maquetacion boton

GuillemBo committed:
dropdown menu

GuillemBo committed:
fix styles, add logout svg

GuillemBo committed:
change svg img

GuillemBo committed:
Language logout texts

GuillemBo committed:
tests implementation

GuillemBo committed:
apply the new features on mobile-nav component

mcallejo-10 reviewed (approved):
Looks good to me!

ivilarop commented:
Wouldn't it have been better to implement the functionality and then the button?

ivilarop commented on src/app/core/layout/header/desktop-nav/desktop-nav.component.ts:
```diff
@@ -1,25 +1,54 @@
   }
 
+  ngOnInit(): void {
+    this.getUserFromLocalStorage()
```
Why from local storage? As I mentioned in the PR for task 195, the username should be retrieved from the token you'll have in the encrypted cookie. You'll need to run 195 first and then retrieve the username.

ivilarop commented on src/app/core/layout/header/desktop-nav/desktop-nav.component.ts:
```diff
@@ -1,25 +1,54 @@
     this.isLoggedIn = isLogged
   }
+
+  getUserFromLocalStorage (): void {
```
Why from local storage? As I mentioned in the PR for task 195, the username should be retrieved from the token you'll have in the encrypted cookie. You'll need to run 195 first and then retrieve the username.

ivilarop commented on src/app/core/layout/header/mobile-nav/mobile-nav.component.spec.ts:
```diff
@@ -72,4 +72,68 @@ describe('MobileNavComponent', () => {
+    document.body.removeChild(userBtn);
+  });
+
+  it('should load user from localStorage', () => {
```
modify it when you have recovered the username from the encrypted cookie

ivilarop commented on src/app/core/layout/header/mobile-nav/mobile-nav.component.spec.ts:
```diff
@@ -72,4 +72,68 @@ describe('MobileNavComponent', () => {
+    localStorage.removeItem('username');
+  });
+
+  it('should set user to empty string if no username in localStorage', () => {
```
modify it when you have recovered the username from the encrypted cookie

ivilarop commented on src/app/core/layout/header/mobile-nav/mobile-nav.component.ts:
```diff
@@ -1,25 +1,54 @@
   }
 
+  ngOnInit(): void {
+    this.getUserFromLocalStorage()
```
the same as on desktop

ivilarop commented on src/app/core/layout/header/mobile-nav/mobile-nav.component.ts:
```diff
@@ -1,25 +1,54 @@
     this.isLoggedIn = isLogged
   }
+
+  getUserFromLocalStorage (): void {
```
the same as on desktop

GuillemBo committed:
get username from token

GuillemBo committed:
Merge branch 'develop' into feature/186-dropdown-menu-logout

GuillemBo committed:
merge

GuillemBo committed:
fix tests

GuillemBo committed:
fix nav.service test conflict

ivilarop commented on src/app/services/auth.service.ts:
```diff
@@ -20,15 +22,23 @@ export class AuthService {
+  }
+
   // Method to update the user role when authentication changes
   updateUserRoleFromToken(): void {
```
The name updateUserRoleFromToken() is short and can be misleading, since it not only updates the role, but also the username.

GuillemBo committed:
fix requested changes, fix tests since conflicts with new isloggedIn function

GuillemBo committed:
Merge branch 'develop' into feature/186-dropdown-menu-logout

GuillemBo commented:
In this pull request, I implemented a dropdown menu that includes a logout button and displays the username of the logged-in user. To achieve this, I modified authService and the updateUserRoleFromToken() function to also return the username, as it previously only returned the user role. Given this new functionality, I renamed the function to updateUserRoleAndUsernameFromToken().

Additionally, I updated mobile-nav.component.ts to align with desktop-nav.component.ts, as I had previously overlooked this adjustment in feature/195.
